### PR TITLE
Downgrade several 7.0.* packages to 6.0.*

### PR DESCRIPTION
**Pull request by stephentoub**
This pains me, as it's a technical step backwards.  But there are two problems today:
1) The .NET support policy today unfortunately applies not only to the .NET SDK and runtime but also to nuget packages released in the same wave, which means that if someone is relying on eg System.Text.Json and is running on .NET 6, having SK depend on System.Text.Json with a minimum version of 7.0 forces the consumer to also use the 7.0 version of System.Text.Json from the nuget, which then means that one assembly falls under the STS support policy instead of LTS support policy, which means that one assembly will have support end for it a few months earlier than it otherwise would.  I've raised this issue for further evaluation on the .NET side of things, but in the meantime, this lowers the version number to remove the perceived problem and possible barrier to adoption.
2) Azure Functions has two deployment models: in-process and isolated.  Ideally functions use isolated, which gives them the freedom to reference whatever they need.  The in-process model is exactly what it sounds like: the function runs in the same process as the host, and the host pins several dependencies at 6.0 versions.  That means that if a function references a 7.0 version, it'll fail to load in the in-process model.  While we'd like for folks to be using the isolated model, we can't force it, and we don't want a need for the in-process model to block SK usage.

This commit downgrades back to the 6.0 versions, at least where possible.  Some of the connectors reference libraries (e.g. NRedisStack, pgvector, etc.) that themselves have a 7.0 dependency.

Closes https://github.com/microsoft/semantic-kernel/issues/1793

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @ericstj

File: dotnet/src/Connectors/Connectors.UnitTests/Connectors.UnitTests.csproj [hunk 1]
```diff
@@ -20,6 +20,10 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
+
+    <!-- Because some of the referenced projects have dependencies that themselves have System.Text.Json set with a minimum of 7.0. -->
```
Comment (ericstj) [inline dotnet/src/Connectors/Connectors.UnitTests/Connectors.UnitTests.csproj:24]: In those cases can't you just let those references win?  Why do you need an additional reference here?

Comment (stephentoub) [reply dotnet/src/Connectors/Connectors.UnitTests/Connectors.UnitTests.csproj:24]: Because otherwise it produces nuget conflicts warnings

**Comment by shawncal** [comment]
Could you please also add the downgraded packages to [ignore](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore) sections in https://github.com/microsoft/semantic-kernel/blob/main/.github/dependabot.yml?
Otherwise, we'll be right back where we are in a couple days ;p

**Comment by stephentoub** [comment]
> Could you please also add the downgraded packages to [ignore](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#ignore) sections

I tried.  Let me know if this looks right / wrong.

**Review by shawncal — APPROVED** [review]

**Comment by Woland2k** [comment]
Thank you for the update and listening for the community on this!

**Pull request merged**